strawberryfields/backends/tfbackend/circuit.py

```diff
@@ -612,161 +551,147 @@ def measure_fock(self, modes, select=None, **kwargs):
                     raise ValueError("'select' must be have the same shape as 'modes'")
 
         # carry out the operation
-        with self.graph.as_default():
-            evaluate_results, session, feed_dict, close_session = ops._check_for_eval(kwargs)
-            num_reduced_state_modes = len(modes)
-            reduced_state = self._state
-            if self._state_is_pure:
-                mode_size = 1
-            else:
-                mode_size = 2
-            if self._batched:
-                batch_size = self._batch_size
-                batch_offset = 1
-            else:
-                batch_size = 1
-                batch_offset = 0
+        num_reduced_state_modes = len(modes)
```

Comment:
Following up on the previous comment: this part could now be extracted e.g. as `apply_operation`? And it could return `new_state` to integrate it into the logic. Having a docstring could also help out with further details on the logic.